src/shipit_static_analysis/shipit_static_analysis/workflow.py

```diff
@@ -126,57 +125,36 @@ def run(self, revision):
             # Update to the target revision
             self.hg.update(rev=revision.mercurial, clean=True)
 
-            # Get the parents revisions
-            parent_rev = 'parents({})'.format(revision.mercurial)
-            parents = self.hg.identify(id=True, rev=parent_rev).decode('utf-8').strip()
-
-            # Find modified files by this revision
-            modified_files = []
-            for parent in parents.split('\n'):
-                changeset = '{}:{}'.format(parent, revision.mercurial)
-                status = self.hg.status(change=[changeset, ])
-                modified_files += [f.decode('utf-8') for _, f in status]
-            logger.info('Modified files', files=modified_files)
-
-        # List all modified lines from current revision changes
-        patch = Patch.parse_patch(
-            self.hg.diff(change=revision.mercurial, git=True).decode('utf-8'),
-            skip_comments=False,
-        )
-        modified_lines = {
-            # Use all changes in new files
-            filename: diff.get('touched', []) + diff.get('added', [])
-            for filename, diff in patch.items()
-        }
-
-        with stats.api.timer('runtime.mach'):
-            # mach configure with mozconfig
-            logger.info('Mach configure...')
-            run_check(['gecko-env', './mach', 'configure'], cwd=self.repo_dir)
-
-            # Build CompileDB backend
-            logger.info('Mach build backend...')
-            cmd = ['gecko-env', './mach', 'build-backend', '--backend=CompileDB']
-            run_check(cmd, cwd=self.repo_dir)
-
-            # Build exports
-            logger.info('Mach build exports...')
-            run_check(['gecko-env', './mach', 'build', 'pre-export'], cwd=self.repo_dir)
-            run_check(['gecko-env', './mach', 'build', 'export'], cwd=self.repo_dir)
+            # Analyze files in revision
+            revision.analyze_files(self.hg)
+
+        # Only run mach if revision has any C/C++ files
+        if revision.has_clang_files:
+            with stats.api.timer('runtime.mach'):
+                # mach configure with mozconfig
+                logger.info('Mach configure...')
+                run_check(['gecko-env', './mach', 'configure'], cwd=self.repo_dir)
+
+                # Build CompileDB backend
+                logger.info('Mach build backend...')
+                cmd = ['gecko-env', './mach', 'build-backend', '--backend=CompileDB']
+                run_check(cmd, cwd=self.repo_dir)
+        else:
+            logger.info('No clang files detected, skipping mach')
```

Comment:
We should monitor and check if we start seeing errors related to mach. In theory, there could be issues about the creation of the virtualenv (since it might happen from multiple processes at the same time in lint.py).
If we do start to see errors, we can just run `./mach python --version` here to ensure the virtualenv is created.